atomate/utils/database.py

```diff
@@ -125,11 +125,18 @@ def from_db_file(cls, db_file, admin=True):
             user = creds.get("readonly_user")
             password = creds.get("readonly_password")
 
-        kwargs = {}
+        kwargs = dict(creds)
         if "authsource" in creds:
             kwargs["authsource"] = creds["authsource"]
         else:
             kwargs["authsource"] = creds["database"]
-
+        if 'ssl' in creds:
+            kwargs['ssl'] = bool(creds['ssl'])
+
+        # Remove keys which may cause collision when calling MongoClient
+        for key in list(kwargs.keys()):
```

Comment:
Could you explain the rationale for this?

Are you trying to avoid a specific problem that you've seen in practice happen, or are you simply trying to be extra cautious?

I would generally avoid this; for example, let's say that MongoClient in a later version of pymongo supports some new kwarg based on some new feature of MongoDB. A user supplies that new kwarg, but it fails due to this line of code. The user is then confused - is atomate restricting the keys for some scientific reason (e.g., atomate will break if the new kwarg is supplied) or will it be OK to add the new kwarg to the list of allowed kwargs?

Let me know if there's some specific reason this line needs to be in there, thanks!

Reply:
Nevermind I see the point now - you want to avoid having things like ``readonly_user`` and ``readonly_password`` from creds passed into kwargs. I think there should be a better solution to this though